File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
from collections import deque
import sys
input = sys.stdin.readline

n,m,v=map(int, input().split())
grapgh=[[0]*(n+1) for _ in range(n+1)]

for _ in range(m):
    a,b=map(int, input().split())
    grapgh[a][b]=1
    grapgh[b][a]=1


# 간선이 아니라 점으로 방문기록
visit1=[0]*(n+1) # DFS 방문기록
visit2=[0]*(n+1)  # BFS 방문기록

def DFS(arr,v):
    visit1[v]=1
    print(v, end=' ')
    for i in range(1, n+1):
        if arr[v][i]==1 and visit1[i]==0:
            visit1[i]=1
            DFS(arr,i)
            
def BFS(arr,v):
    visit2[v]=1
    queue=deque()
    queue.append(v)
    while queue:
        v=queue.popleft()
        print(v, end=' ')
        for i in range(1, n+1):
            if grapgh[v][i]==1 and visit2[i]==0:
                visit2[i]=1
                queue.append(i)
    

DFS(grapgh, v)
print()
BFS(grapgh, v)

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.cc
#include <iostream>
#include <queue>

using namespace std;
#define MAX 1001


int N, M, V;
int arr[MAX][MAX];
bool visited[MAX] ={ 0,};
queue<int> q;

void reset(){
    for(int i=1; i<=N; i++){
        visited[i]=0;
    }
}

void DFS(int v){
    visited[v]=true;
    cout<<v<<" ";

    for(int i=1; i<=N; i++){
        if(arr[v][i]==1 && visited[i]==false){
            DFS(i);
        }
    }
}

void BFS(int v){
    q.push(v);
    visited[v]=true;
    cout<<v<<" ";

    while(!q.empty()){
        v=q.front();
        q.pop();

        for(int i=1; i<=N; i++){
            if(arr[v][i]==1 && visited[i]==false){
                q.push(i);
                visited[i]=true;
                cout<<i<<" ";
            }
        }
    }
}



int main() {
    cin>> N>>M>>V;
    for(int i=0; i<M; i++){
        int a, b;
        cin>>a>>b;
        arr[a][b] = 1;
        arr[b][a] =1 ;
    }

    reset();
    DFS(V);

    cout<<'\n';

    reset();
    BFS(V);
}